File: src/IntersectionOfTwoArrays.java
import java.util.Arrays;
import java.util.HashSet;

public class IntersectionOfTwoArrays {

  public static void main(String[] args) {
    int[] nums1 = {1, 2, 2, 1};
    int[] nums2 = {2, 2};
    int[] res = intersection(nums1, nums2);
    System.out.println(Arrays.toString(res));
  }

  public static int[] intersection(int[] nums1, int[] nums2) {
    HashSet<Integer> container = new HashSet<>();
    HashSet<Integer> output = new HashSet<>();

    for (int num : nums1) {
      container.add(num);
    }
    for (int num : nums2) {
      if (container.contains(num)) {
        output.add(num);
      }
    }

    int[] outputArray = new int[output.size()];
    int index = 0;
    for (int x : output) {
      outputArray[index] = x;
      index++;
    }

    return outputArray;
  }
}

File: src/javascript/IntersectionOfTwoArrays.js
const intersection = function (nums1, nums2) {
    let set1 = new Set(nums1);
    let set2 = new Set(nums2);
    let result = [];
    for (let num of set1) {
        if (set2.has(num)) {
            result.push(num);
        }
    }
    return result;
};

let nums1 = [1, 2, 2, 1], nums2 = [2, 2];
console.log(intersection(nums1, nums2));